Support Recursive JSON Type

**Is your feature request related to a problem? Please describe.**
Support Recursive JSON Type.

**Describe the solution you'd like**
Types like this work
```
type JSONContent {
  type String
  content JSONContent[]?
  text String?
}
```

**Describe alternatives you've considered**
n/a

**Additional context**
Currently this type creates an error:
```
error TS7022: 'JSONContentSchema' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.

5 export const JSONContentSchema = z.object({
               ~~~~~~~~~~~~~~~~~

: Error compiling generated code
```

--- a/packages/sdk/src/code-gen.ts
+++ b/packages/sdk/src/code-gen.ts
@@ -15,6 +15,7 @@ export function createProject(options?: CompilerOptions) {
             strict: true,
             skipLibCheck: true,
             noEmitOnError: true,
+            noImplicitAny: false,
             ...options,
         },
     });

--- a/tests/integration/tests/enhancements/json/crud.test.ts
+++ b/tests/integration/tests/enhancements/json/crud.test.ts
@@ -344,4 +344,49 @@ describe('Json field CRUD', () => {
         expect(u2.profile.ownerId).toBe(2);
         expect(u2.profile.nested.userId).toBe(3);
     });
+
+    it('works with recursive types', async () => {
+        const params = await loadSchema(
+            `
+            type Content {
+                type String
+                content Content[]?
+                text String?
+            }
+            
+            model Post {
+                id Int @id @default(autoincrement())
+                content Content @json
+                @@allow('all', true)
+            }
+            `,
+            {
+                provider: 'postgresql',
+                dbUrl,
+            }
+        );
+
+        prisma = params.prisma;
+        const db = params.enhance();
+        const post = await db.post.create({
+            data: {
+                content: {
+                    type: 'text',
+                    content: [
+                        {
+                            type: 'text',
+                            content: [
+                                {
+                                    type: 'text',
+                                    text: 'hello',
+                                },
+                            ],
+                        },
+                    ],
+                },
+            },
+        });
+
+        await expect(post.content.content[0].content[0].text).toBe('hello');
+    });
 });

--- a/tests/integration/tests/enhancements/json/typing.test.ts
+++ b/tests/integration/tests/enhancements/json/typing.test.ts
@@ -332,4 +332,57 @@ async function main() {
             }
         );
     });
+
+    it('supports recursive definition', async () => {
+        await loadSchema(
+            `
+            type Content {
+                type String
+                content Content[]?
+                text String?
+            }
+            
+            model Post {
+                id Int @id @default(autoincrement())
+                content Content @json
+            }
+            `,
+            {
+                provider: 'postgresql',
+                pushDb: false,
+                compile: true,
+                extraSourceFiles: [
+                    {
+                        name: 'main.ts',
+                        content: `
+import type { Content } from '.zenstack/models';
+import { enhance } from '.zenstack/enhance';
+import { PrismaClient } from '@prisma/client';
+
+async function main() {
+    const content: Content = {
+        type: 'text',
+        content: [
+            {
+                type: 'text',
+                content: [
+                    {
+                        type: 'text',
+                        text: 'hello',
+                    },
+                ],
+            },
+        ],
+    }
+
+    const db = enhance(new PrismaClient());
+    const post = await db.post.create({ data: { content } });
+    console.log(post.content.content?.[0].content?.[0].text);
+}
+                `,
+                    },
+                ],
+            }
+        );
+    });
 });

